src/accelerate/accelerator.py

```diff
@@ -948,7 +951,7 @@ def prepare_model(self, model: torch.nn.Module, device_placement=None):
             if self.mixed_precision == "fp16" and is_torch_version(">=", "1.10"):
                 model.forward = torch.cuda.amp.autocast(dtype=torch.float16)(model.forward)
             elif self.mixed_precision == "bf16" and self.distributed_type != DistributedType.TPU:
-                device_type = "cuda" if torch.cuda.is_available() else "cpu"
+                device_type = "cuda" if self.device.type != "cpu" and torch.cuda.is_available() else "cpu"
```

Comment:
This line (and the next occurrence) can be removed entirely and just pass in `self.device.type` to `torch.autocast`

Reply:
good point. done

Reply:
done